app/templates/pdf/order_invoice.html

```diff
@@ -240,7 +260,12 @@ <h1>{{ _('INVOICE') }}</h1>
             <td></td>
             <td></td>
             <td class="grand total"><b>{{ _('Grand Total') }}</b></td>
-            <td class="grand total"> {{ event.payment_currency }}{{ order.amount | money }}</td>
+            <td class="grand total"> 
+              {% if order.amount %}
```

Comment:
Above you checked order.amount !=0 and here only order.amount, any specific reason?

Reply:
Thank you for editing it. I was not available today but I got the point. It should have been consistent.